lib/diagnostics_channel.js

```diff
@@ -23,11 +29,44 @@ const { triggerUncaughtException } = internalBinding('errors');
 
 const { WeakReference } = internalBinding('util');
 
+function decRef(channel) {
+  channel._weak.decRef();
+  if (channel._weak.getRef() === 0) {
+    delete channels[channel.name];
+  }
+}
+
+function markActive(channel) {
+  // eslint-disable-next-line no-use-before-define
```

Comment:
Isn't just moving this function down the file to avoid this eslint skip?

Reply:
No, because it would have to be defined after it's used which would mean the call locations would then need this eslint disable. The reference is circular, that's the problem.